### PR TITLE
[Feature]: Fetch steam deck comaptibility

### PR DESCRIPTION
**Pull request by kohend**
Fetch steamdeck and protondb compatibiltiy and show it on game page.

image
image
image
image

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [X] Tested the feature and it's working on a current and clean install.
- [X] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [X] Created / Updated Tests (If necessary)
- [X] Created / Updated documentation (If necessary)

### CODE REVIEW

File: src/common/types.ts [hunk 1]
```diff
@@ -603,13 +603,18 @@ export interface ProtonDBCompatibilityInfo {
   level: string
 }
 
+export interface SteamDeckComp {
```
Comment (Nocccer) [inline src/common/types.ts:606]: I know my preference, but would love to combine protondb and this together.

**Review by Nocccer — APPROVED** [review]
Nice. Left some opinion but nothing that holds back this feature.

**Comment by kohend** [comment]
image

File: src/backend/wiki_game_info/wiki_game_info.ts [hunk 2]
```diff
@@ -50,9 +50,10 @@ export async function getWikiGameInfo(
         isMac ? getInfoFromAppleGamingWiki(title) : null
       ])
 
+    const steamID = gamesdb?.steamID ? gamesdb.steamID : ''
```
Comment (Nocccer) [inline src/backend/wiki_game_info/wiki_game_info.ts:53]:
```suggestion
    if (isLinux) {
      const steamID = pcgamingwiki?.steamID ?? gamesdb?.steamID
      const [protondb, steamdeck] = await Promise.all([     
        getInfoFromProtonDB(steamID),
        getSteamDeckComp(steamID)
    ])
    }
```

**Review by Nocccer — CHANGES_REQUESTED** [review]
I don't like the frontend implementation. Something is wrong here.
Maybe not your fault but i would make a new component like HowLonToBeat and Gamescore.
The component combines apple crossover compatiblity and proton/steamdeck compatibility.
If you don't know how, i could do that.

**Comment by Nocccer** [comment]
@arielj @flavioislima Can you look over the frontend implementation? 
Code in the backend is already done and tested.

File: src/frontend/screens/Game/GamePage/index.tsx [hunk 3]
```diff
@@ -19,7 +19,11 @@ import {
   Pause,
   Warning,
   Hardware,
-  Error
+  Error,
+  CheckCircleOutline,
```
Comment (kohend) [inline src/frontend/screens/Game/GamePage/index.tsx:23]: Perhaps the check will be better if it's the full and not the outline version?

File: src/frontend/screens/Game/GamePage/index.tsx [hunk 4]
```diff
@@ -314,12 +317,25 @@ export default React.memo(function GamePage(): JSX.Element | null {
...
+    const hasProtonDB = steamInfo?.compatibilityLevel
+    const hasSteamDeckCompat = steamInfo?.steamDeckCatagory
     const steamLevelNames = [
-      t('info.steam-deck-compat.level0', 'Unknown'),
-      t('info.steam-deck-compat.level1', 'Unsupported'),
-      t('info.steam-deck-compat.level2', 'Playable'),
-      t('info.steam-deck-compat.level3', 'Verified')
+      <HelpOutline
+        key={0}
+        style={{ marginLeft: '5px', cursor: 'not-allowed' }}
```
Comment (kohend) [inline src/frontend/screens/Game/GamePage/index.tsx:325]: Perhaps color these to make them more differentiated?

File: src/backend/wiki_game_info/wiki_game_info.ts [hunk 5]
```diff
@@ -50,19 +50,21 @@ export async function getWikiGameInfo(
...
       ])
 
-    const steamID = gamesdb?.steamID ? gamesdb.steamID : ''
-    const [protondb, steamdeck] = await Promise.all([
-      getInfoFromProtonDB(isLinux ? steamID : ''),
-      getSteamDeckComp(isLinux ? steamID : '')
-    ])
+    let steamInfo = null
+    if (isLinux) {
+      const steamID = pcgamingwiki?.steamID ?? gamesdb?.steamID
```
Comment (kohend) [inline src/backend/wiki_game_info/wiki_game_info.ts:55]: If there's a new default and the gamesdb is a fallback, I think the fallback should be tested as well.

**Review by kohend — COMMENTED** [review]
Looks better than what I did, well, I'm not a UI person

**Review by flavioislima — APPROVED** [review]
I think looks good using the icons on the UI 👍🏽

**Pull request merged**